src/k8s/K8sWorkerController.cpp

```diff
@@ -58,6 +59,26 @@ K8sWorkerController::K8sWorkerController(std::string masterIp, int numberOfWorke
 
 K8sWorkerController::~K8sWorkerController() { delete this->interface; }
 
+static K8sWorkerController *instance = nullptr;
+
+K8sWorkerController *K8sWorkerController::getInstance() {
+    if (instance == nullptr) {
+        throw std::runtime_error("K8sWorkerController is not instantiated");
+    }
+    return instance;
+}
+
+K8sWorkerController *K8sWorkerController::getInstance(std::string masterIp, int numberOfWorkers,
+                                                      SQLiteDBInterface *metadb) {
+    // TODO(thevindu-w): synchronize
+    if (instance == nullptr) {
+        instance = new K8sWorkerController(masterIp, numberOfWorkers, metadb);
+    } else {
+        controller_logger.warn("Not initializing again");
+    }
+    return instance;
```

Comment:
This can lead to a problem such as the test case failure in unit tests.
Consider following instance,
1. Create a metadb object
2. Based on that create k8s worker controller.
3. Delete the created metadb object in step 1.
4. Create a new metadb object.
5. Based on the new metadb object call the getInstance function. The instance returned of the `K8SWorkerController` is still refering to the old metadb object.

This is causing the unit test failure in `TestScaleUpAndDown`.